src/main/java/seedu/guest/commons/util/StringUtil.java

```diff
@@ -38,6 +38,42 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
                 .anyMatch(preppedWord::equalsIgnoreCase);
     }
 
+    /**
+     * Returns true if the {@code sentence} contains the {@code date}.
+     *   Ignores hyphen and space, but a full date match is required.
+     *   <br>examples:<pre>
+     *       containsDateIgnoreHyphenIgnoreSpace("13/09/22 - 15/09/22", "13/09/22 - 15/09/22") == true
+     *       containsDateIgnoreHyphenIgnoreSpace("13/09/22 - 15/09/22", "13/09/22 15/09/22") == true
+     *       containsDateIgnoreHyphenIgnoreSpace("13/09/22 - 15/09/22", "13/09/22") == true
+     *       containsDateIgnoreHyphenIgnoreSpace("13/09/22 - 15/09/22", "15/09/22") == true
+     *       containsDateIgnoreHyphenIgnoreSpace("13/09/22 - 15/09/22", "13/09/22 - 19/09/22")
+     *              == false //not a full date match
+     *       containsDateIgnoreHyphenIgnoreSpace("13/09/22 - 15/09/22", "14/09/22")
+     *              == false //not a full date match
+     *       containsDateIgnoreHyphenIgnoreSpace("13/09/22 - 15/09/22", "13/09")
+     *              == false //not a full date match
+     *       </pre>
+     * @param sentence cannot be null
+     * @param date cannot be null, cannot be empty, must be a single date
+     */
+    public static boolean containsDateIgnoreHyphenIgnoreSpace(String sentence, String date) {
+        requireNonNull(sentence);
+        requireNonNull(date);
+
+        String preppedDate = date.trim().replace("-", "");
+        if (preppedDate.isEmpty()) {
+            return false;
+        }
+        checkArgument(!preppedDate.isEmpty(), "date parameter cannot be empty");
+        checkArgument(preppedDate.split("\\s+").length == 1, "date parameter should be a single date");
```

Comment:
Not sure if this is still needed or can be replaced with assert? Since it would be handled by the `FindCommandParser`

Reply:
I feel that it should be left in for defensive programming purposes should the method be used in other scenarios.

Reply:
Got it!